sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -376,4 +379,45 @@ public static Context mergeContexts(Context into, Context from) {
 
         return returnContext;
     }
+
+    /**
+     * Writes a {@link ByteBuffer} into an {@link OutputStream}.
+     * <p>
+     * This method provides writing optimization based on the type of {@link ByteBuffer} and {@link OutputStream}
+     * passed. For example, if the {@link ByteBuffer} has a backing {@code byte[]} this method will access that directly
+     * to write to the {@code stream} instead of buffering the contents of the {@link ByteBuffer} into a temporary
+     * buffer.
+     *
+     * @param buffer The {@link ByteBuffer} to write into the {@code stream}.
+     * @param stream The {@link OutputStream} where the {@code buffer} will be written.
+     * @throws IOException If an I/O occurs while writing the {@code buffer} into the {@code stream}.
+     */
+    public static void writeByteBufferToStream(ByteBuffer buffer, OutputStream stream) throws IOException {
```

Comment:
CoreUtils has a variety of util methods and may not be easy to find the ones that deal with data. Should we create a new util type for just handling all data transformations?

Reply:
This has been moved into `ImplUtils` and a method added to `FluxUtil` to write an entire `Flux<ByteBuffer>` to `OutputStream`. But I agree in the future we may want to discuss a new utility for data-based operations.